src/SprykerEco/Zed/Unzer/Business/Refund/RefundStrategy/LastShipmentItemExpenseRefundStrategy.php

```diff
@@ -164,4 +146,45 @@ protected function isPaymentUnzerOrderItemAlreadyRefunded(
 
         return false;
     }
+
+    /**
+     * @param array<\Generated\Shared\Transfer\ItemTransfer> $itemTransfers
+     * @param array<int> $salesOrderItemIds
+     *
+     * @return bool
+     */
+    protected function expenseShouldBeRefunded(array $itemTransfers, array $salesOrderItemIds): bool
+    {
+        $itemsCountForRefund = 0;
+        foreach ($itemTransfers as $itemTransfer) {
+            if (in_array($itemTransfer->getIdSalesOrderItemOrFail(), $salesOrderItemIds, true)) {
+                $itemsCountForRefund++;
+
+                continue;
+            }
+
+            if ($itemTransfer->getCanceledAmountOrFail() !== 0) {
+                $itemsCountForRefund++;
+            }
+        }
+
+        return $itemsCountForRefund === count($itemTransfers);
+    }
+
+    /**
+     * @param array<\Generated\Shared\Transfer\ItemTransfer> $itemTransfers
+     *
+     * @return bool
+     */
+    protected function allItemsAreRefunded(array $itemTransfers): bool
+    {
+        $refundedItemsCount = 0;
+        foreach ($itemTransfers as $itemTransfer) {
+            if ($itemTransfer->getCanceledAmountOrFail() !== 0) {
+                $refundedItemsCount++;
```

Comment:
Basically, you don't need to count refunded items here, just return `false` if any item's canceled amount === 0.

Reply:
I need to check if all items are refunded but not any